### ISSUE TITLE
Enhancement request: Fractional second test duration (-t)

### ISSUE THREAD

**Issue by zhibindaxia** [esnet/iperf#1007]
Is -t parameter can only accept integers?
When I use
server: iperf3 -s -p 5001 -i 0.1 &
client: iperf3 -c 10.0.0.1 -p 5001 -i 0.1 -t 0.1
The result is a endless stream

**Comment by bmah888** [comment]
Right, the `-t` flag only takes integer parameters.  Is there actually a use case for accepting floating-point test durations?

So in the sample command-line above, the `-t 0.1` is interpreted as `-t 0`, which basically means to run forever.  Upon looking, I don't see anything in the documentation that covers this fact, so we probably ought to add some words for this.

**Comment by zhibindaxia** [comment]
Yes, I need floating-point tests to shorten the experimental sampling time (such as sampling once per rtt).


Better experimental results can be achieved



------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Bruce A. Mah"<notifications@github.com&gt;;
发送时间:&nbsp;2020年5月29日(星期五) 凌晨0:51
收件人:&nbsp;"esnet/iperf"<iperf@noreply.github.com&gt;;
抄送:&nbsp;"彭智彬"<814976254@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [esnet/iperf] On the problem of parameter -t (#1007)





 
Right, the -t flag only takes integer parameters.  Is there actually a use case for accepting floating-point test durations?
 
So in the sample command-line above, the -t 0.1 is interpreted as -t 0, which basically means to run forever.  Upon looking, I don't see anything in the documentation that covers this fact, so we probably ought to add some words for this.

**Comment by bmah888** [comment]
It sounds like what you really want is fractional-second measurement intervals (`-i` flag), which are indeed supported.  I'm not convinced that a floating-point test duration `-t` is really useful, and while it's not conceptually hard to do, making this change has some consequences in other parameters as well (`-O` flag, changing variable types within different data structures, ensuring backward compatibility between different iperf3 versions, etc.).
